aten/src/THCUNN/SpatialClassNLLCriterion.cu

```diff
@@ -121,10 +121,12 @@ __global__ void cunn_SpatialClassNLLCriterion_updateOutput_kernel(
 template<typename T>
 __global__ void cunn_SpatialClassNLLCriterion_sizeAverage_kernel(
           T *output,
-          T *total_weight)
+          T *total_weight,
+          int nElement)
 {
-  if (*total_weight > 0)
+  if ((*total_weight != 0 && output != 0) || nElement == 0) {
```

Comment:
Ditto, this is getting complicated. The downside here is that it'll be difficult to maintain in the future -- why do we need to do this?

Reply:
It is what you asked me to fix in [previous review](https://github.com/pytorch/pytorch/pull/31491#discussion_r365461289)